scripts/irods/upgrade_configuration.py

```diff
@@ -92,6 +92,36 @@ def upgrade(irods_config):
                                 'Please resolve this naming conflict manually and try again.' % (old_path, new_path, new_path))
                     shutil.move(old_path, new_path)
 
+    if old_version_tuple < (4, 3, 0):
+        # Load server_config.json.
+        with open(paths.server_config_path()) as f:
+            server_config = json.load(f)
+
+        # Set default values for the hosts_config.json and host_access_control_config.json.
+        server_config['host_access_control'] = {'access_entries': []}
+        server_config['host_resolution'] = {'host_entries': []}
+
+        # Merge hosts_config.json into server_config.json if it is available.
+        if os.path.exists(paths.hosts_config_path()):
+            with open(paths.hosts_config_path()) as f:
+                hosts_config = json.load(f)
+
+            if 'host_entries' in hosts_config and len(hosts_config['host_entries']) > 0:
+                # Ignores the schema information in the JSON file.
+                server_config['host_resolution']['host_entries'] = hosts_config['host_entries']
```

Comment:
this is where we would swizzle to match a new schema

Reply:
Yes.

Reply:
We now swizzle the contents of host_entries to match the v4 schema.